apps/mobile/src/hooks/useRegisterForNotifications.ts

```diff
@@ -0,0 +1,272 @@
+import { useCallback, useState } from 'react'
+import { Wallet } from 'ethers'
+import { useAuthGetNonceV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/auth'
+import FCMService from '@/src/services/notifications/FCMService'
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks'
+import {
+  toggleAppNotifications,
+  updateLastTimePromptAttempted,
+  updatePromptAttempts,
+} from '@/src/store/notificationsSlice'
+import Logger from '@/src/utils/logger'
+
+import { useDefinedActiveSafe } from '@/src/store/hooks/activeSafe'
+import { useGTW } from './useGTW'
+import { addOrUpdateDelegatedAccount, selectDelegatedAccounts } from '../store/delegatedSlice'
+import { Address, SafeInfo } from '../types/address'
+import { useNotificationPayload } from './useNotificationPayload'
+import { ERROR_MSG } from '../store/constants'
+import { getSigner } from '../utils/notifications'
+import { useNotificationGTWPermissions } from './useNotificationGTWPermissions'
+
+type RegisterForNotificationsProps = {
+  loading: boolean
+  error: string | null
+}
+
+interface NotificationsProps {
+  registerForNotifications: () => Promise<RegisterForNotificationsProps>
+  unregisterForNotifications: () => Promise<RegisterForNotificationsProps>
+  updatePermissionsForNotifications: () => Promise<RegisterForNotificationsProps>
+  isLoading: boolean
+  error: string | null
+}
+
+const useRegisterForNotifications = (): NotificationsProps => {
+  // Local states
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+  // Custom hooks
+  const { data: nonceData } = useAuthGetNonceV1Query()
+  const { registerForNotificationsOnBackEnd, unregisterForNotificationsOnBackEnd } = useGTW()
+  const { getNotificationRegisterPayload } = useNotificationPayload()
+  // Redux
+  const dispatch = useAppDispatch()
+  const activeSafe = useDefinedActiveSafe()
+  const delegatedAccounts = useAppSelector(selectDelegatedAccounts)
+  const { ownerFound, accountType } = useNotificationGTWPermissions().getAccountType()
+  /*
+   * Push notifications can be enabled by an two type of users. The owner of the safe or an observer of the safe
+   * In the first case, the owner can subscribe to ALL NotificationTypes listed in @safe-global/store/gateway/AUTO_GENERATED/notifications
+   * including confirmation requests notifications
+   * In the second case, the observer notifications will not include confirmation requests
+   *
+   * We only notify required confirmation events to owners or delegates
+   * to prevent other subscribers from receiving "private" events
+   */
+  const registerForNotifications = useCallback(async () => {
+    try {
+      setLoading(true)
+      setError(null)
+      // Step 1 - Set up Firebase Cloud Messaging steps
+      await FCMService.registerAppWithFCM()
+      await FCMService.saveFCMToken()
+      const fcmToken = await FCMService.getFCMToken()
+      FCMService.listenForMessagesBackground()
```

Comment:
if you create a function `setUpFirebaseCloudMessaging` and move this code to it, there won't be a need for the comment as the title of the function already explains what is going on.